appendix/stata-guide.tex

```diff
@@ -177,127 +216,204 @@ \subsection{Writing loops}
+and always consider whether missing values will affect the evaluation conditional expressions.
 
 \codeexample{stata-conditional-expressions1.do}{./code/stata-conditional-expressions1.do}
 
-You should also always use \texttt{if-else} statements when applicable even if you can express the same
-thing with two separate \texttt{if} statements. When using \texttt{if-else} statements you are
-communicating to anyone reading your code that the two cases are mutually exclusive in an \texttt{if-else} statement
-which makes your code more readable. It is also less error-prone and easier to update.
+Use \texttt{if-else} statements when applicable
+even if you can express the same thing with two separate \texttt{if} statements.
+When using \texttt{if-else} statements you are communicating to anyone reading your code
+that the two cases are mutually exclusive which makes your code more readable.
+It is also less error-prone and easier to update if you want to change the condition.
 
 \codeexample{stata-conditional-expressions2.do}{./code/stata-conditional-expressions2.do}
 
 \subsection{Using macros}
 
-Stata has several types of \textbf{macros} where numbers or text can be stored temporarily, but the two most common
-macros are \textbf{local} and \textbf{global}. Locals should always be the default type and globals should only
-be used when the information stored is used in a different do-file. Globals are error-prone since they are
-active as long as Stata is open, which creates a risk that a global from one project is incorrectly used in
-another, so only use globals where they are necessary. Our recommendation is that globals should only be defined in the \textbf{master do-file}.
-All globals should be referenced using both the the dollar sign and the curly brackets around their name;
+Stata has several types of \textbf{macros} where numbers or text can be stored temporarily,
+but the two most common macros are \textbf{local} and \textbf{global}.
+All macros should be defined using the \texttt{=} operator.
+Never abbreviate the commands for \textbf{local} and \textbf{global}.
+Locals should always be the default type and globals should only
+be used when the information stored is used in a different do-file.
+Globals are error-prone since they are active as long as Stata is open,
+which creates a risk that a global from one project is incorrectly used in another,
+so only use globals where they are necessary.
+Our recommendation is that globals should only be defined in the \textbf{master do-file}.
+All globals should be referenced using both the the dollar sign and curly brackets around their name (\texttt{\$\{\}});
 otherwise, they can cause readability issues when the endpoint of the macro name is unclear.
 
+You should use descriptive names for all macros (up to 32 characters; prefer fewer).
+Simple prefixes are useful and encouraged such as \texttt{thisParam}, \texttt{allParams},
+\texttt{theLastParam}, \texttt{allParams}, or \texttt{nParams}.
```

Comment:
I think this is a matter of personal preference. I agree `lastParam` is a useful name, but `thisParam` is not. And I know Ben likes to use `theLocal`, and it may be cute, but it's not strictly necessary. Either way, `allParams` is listed twice.

Reply:
should we offer more general rules on case, ie "any naming style so long as consistent"? But `the_underscore_style` uses up characters fast. And otherwise things shouldn't be in caps right. What do you usually do?